### PR TITLE
index application for donation voting

### PR DESCRIPTION
**Pull request by thelostone-mc**
- Ensures DonationVoting applications are indexed
- Reverts when round is not found

image

Fixes https://github.com/gitcoinco/grants-stack/issues/2856

### CODE REVIEW

File: src/indexer/allo/v2/handleEvent.ts [hunk 1]
```diff
@@ -19,6 +19,51 @@ function generateRoundRoles(poolId: bigint) {
...
+        // { name: 'pointer', type: 'string' },
+      
+    ],
+    values[0]
+  )
+  console.log(values);
+  return shit;
+}
+
+// let shit = decodeEncodedApplicationData(
```
Comment (boudra) [inline src/indexer/allo/v2/handleEvent.ts:48]: nice

File: src/indexer/allo/v2/handleEvent.ts [hunk 2]
```diff
@@ -352,6 +410,68 @@ export async function handleEvent(
...
+        throw new ClientError("Project not found", 404);
+      }
+
+      const encodedData = event.params.data;
+
+      const strategyAddress = parseAddress(event.address);
+      const round  = await db.getRoundByStrategyAddress(chainId, strategyAddress);
+
+      if (!round) {
+        throw new ClientError("Round not found", 404);
```
Comment (boudra) [inline src/indexer/allo/v2/handleEvent.ts:429]:
```suggestion
        throw new Error("Round not found");
```

client error is part of the HTTP API

File: src/index.ts [hunk 3]
```diff
@@ -228,7 +228,7 @@ async function main(): Promise<void> {
       databaseConnectionPool,
       config.databaseSchemaName,
       {
-        watchPg: false,
+        watchPg: true,
```
Comment (boudra) [inline src/index.ts:231]:
```suggestion
        watchPg: false,
```

**Comment by thelostone-mc** [comment]
> are you planning on indexing status updates in another PR?

Yes we'll do it before the review recipients changes

**Pull request merged**